lib/src/client/connection.dart

```diff
@@ -91,7 +92,7 @@ class ClientConnection {
       _userAgent,
     ]);
     metadata?.forEach((key, value) {
-      headers.add(new Header.ascii(key, value));
+      headers.add(new Header(utf8.encode(key), utf8.encode(value)));
```

Comment:
Maybe we should stick with `ascii.encode(key)` for the key?

Reply:
Yes - I missed reverting this one!